docs: added java migration docs (#168)
redhat-developer/app-services-sdk-core: migration/java.md

# Migrating a Java project

> In this example project we use the kafka managment SDK to create kafkas. Each snippet is split into the old SDK and the new updated Kiota SDK.

Reviewer: ```suggestion
In this example project we use the kafka management SDK to create kafkas. Each snippet is split into the old SDK and the new updated Kiota SDK.
```

## Migrating the pom.xml
### Old SDK
```xml
<dependencies>
    <dependency>
        <groupId>com.redhat.cloud</groupId>
        <artifactId>kafka-management-sdk</artifactId>
        <version>0.31.0</version>
        <scope>compile</scope>
    </dependency>
</dependencies>
```

### Kiota SDK

> First add the sontatype remote repoistory so the project can access the required dependencies.

Reviewer: ```suggestion
First add the sonatype remote repository so the project can access the required dependencies.
```

```xml
<repositories>
    <repository>
        <id>ossrh</id>
        <url>https://oss.sonatype.org/service/local/staging/deploy/maven2/</url>
    </repository>
</repositories>
```

Next add properties for the versions of the dependencies.
```xml
<properties>
    <sdk.version>1.0.1</sdk.version>
    <kiota.version>0.3.2</kiota.version>
    <auth.version>0.0.3</auth.version>
</properties>
```

> Now add the required dependencies, this not only adds the kafka managment SDK but also the authentication dependency and the required Kiota dependencies for the SDK.

Reviewer: ```suggestion
Now add the required dependencies, this not only adds the kafka management SDK but also the authentication dependency and the required Kiota dependencies for the SDK.
```

```xml

> <dependencies>

Reviewer: Not sure why we added this in the beginning, but I am feeling we can remove few of the kiota deps, wdyt?

Author: Yeah I am not sure which ones we need, they were just the ones in the docs that were there. Honestly, I think it might actually be good to keep them all as it shows the ones they will use in the future even though they are not required for this small project. WDYT?

Reviewer: Makes sense, we can keep them here for now.
